server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -367,13 +367,31 @@ class DeltaSharedTable(
       format = model.Format(),
       schemaString = cleanUpTableSchema(snapshot.metadataScala.schemaString),
       configuration = getMetadataConfiguration(snapshot.metadataScala.configuration),
-      partitionColumns = snapshot.metadataScala.partitionColumns
+      partitionColumns = snapshot.metadataScala.partitionColumns,
+      version = start
     )
     actions.append(modelProtocol.wrap)
     actions.append(modelMetadata.wrap)
 
     // Third: get files
-    val (changeFiles, addFiles, removeFiles) = cdcReader.queryCDF(start, end, latestVersion)
+    val (changeFiles, addFiles, removeFiles, metadatas) = cdcReader.queryCDF(
+      start, end, latestVersion)
+    metadatas.foreach { cdcDataSpec =>
+      cdcDataSpec.actions.foreach { action =>
+        val metadata = action.asInstanceOf[Metadata]
+        val modelMetadata = model.Metadata(
+          id = metadata.id,
+          name = metadata.name,
+          description = metadata.description,
+          format = model.Format(),
+          schemaString = cleanUpTableSchema(metadata.schemaString),
+          configuration = getMetadataConfiguration(metadata.configuration),
+          partitionColumns = metadata.partitionColumns,
+          version = cdcDataSpec.version
+        )
+        actions.append(modelMetadata.wrap)
```

Comment:
Question: If we always return metadata actions for CDF queries, would it break older clients?

Reply:
Yes it will. 

There are 3 options:
Option 1: The current PR, client and server are both changed, and will be released together. New OSS server will break old client. But this could be resolved by letting the client upgrade to the latest release.

Option 2: Check http header user_agent to return metadata only for streaming query. This is the Databricks delta sharing server behavior. I'm not doing it here because I don't know how to get the httpRequest here: https://sourcegraph.com/github.com/delta-io/delta-sharing/-/blob/server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala?L349

Option 3: Add a parameter in the http request which serve as the same purpose as http user_agent. So the server can be compatible with the older client.

I prefer option 1 or option 2. If I got enough time I would go for option 2. For now I feel option 1 is ok because it only break old clients with new oss server and there's a way to work around it. WDYT?

Reply:
Updated the PR with option 2. Now old client is compatible with new server.